src/main/java/az/gdg/msteam/service/impl/MemberServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+
+    MemberRepository memberRepository;
+    private boolean isValid;
+
+    public MemberServiceImpl(MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
+    }
+
+    @Override
+    public List<MemberDto> getAllMembers() {
+        List<Member> members = new ArrayList<>();
+
+        for (Member member : memberRepository.findAll()) {
+            members.add(member);
+        }
+
+        return MemberMapper.INSTANCE.entityToDtoList(members);
+    }
+
+    @Override
+    public String createMember(MemberDto memberDto) {
+        isValid = true;
+        String message = isMemberValid(memberDto);
+        if (isValid && findByEmail(memberDto.getEmail()) == null) {
+            memberRepository.save(MemberMapper.INSTANCE.dtoToEntity(memberDto));
+            message = "Member is created";
+        } else {
+            message = "Email already exist";
+        }
+        return message;
+    }
+
+    @Override
+    public Member findByEmail(String email) {
+        return memberRepository.findByEmail(email);
+    }
+
+    @Override
+    public String deleteMember(Integer id) {
+        MemberResponse response = new MemberResponse();
+        if (findById(id) != null) {
+            memberRepository.deleteById(id);
+            response.setMessage("Member deleted");
+        } else {
+            response.setMessage("Id is not available");
+        }
+        return response.getMessage();
+    }
+
+    @Override
+    public Member findById(Integer id) {
+        Member member = null;
+        try {
+            member = memberRepository.findById(id).orElseThrow(() -> new MemberNotFoundException("Member doesn't exist with this id"));
+        } catch (MemberNotFoundException e) {
+            e.printStackTrace();
+        }
+        return member;
+    }
+
+    @Override
+    public String updateMember(Integer id, MemberDto memberDto) {
+        Member member = findById(id);
+        String message = "";
+        if (member != null) {
+            isValid = true;
+            message = isMemberValid(memberDto);
+            if (isValid) {
+                member.setFirstName(memberDto.getFirstName());
+                member.setLastName(memberDto.getLastName());
+                member.setEmail(memberDto.getEmail());
+                member.setLinkedin(memberDto.getLinkedin());
+                member.setGithub(memberDto.getGithub());
+                member.setPosition(memberDto.getPosition());
+                member.setPhoto(memberDto.getPhoto());
+                memberRepository.save(member);
+                message = "Member is updated";
+            }
+        } else {
+            message = "Member doesn't exist";
+        }
+        return message;
+    }
+
+    @Override
+    public MemberDto getMemberById(Integer id) {
+        return MemberMapper.INSTANCE.entityToDto(findById(id));
+    }
+
+    private String isMemberValid(MemberDto memberDto) {
```

Comment:
I have validation example in one or two of my repositories, please change it according to my examples.